rasa/nlu/classifiers/logistic_regression_classifier.py

```diff
@@ -0,0 +1,190 @@
+
+
+@DefaultV1Recipe.register(
+    DefaultV1Recipe.ComponentType.INTENT_CLASSIFIER, is_trainable=True
+)
+class LogisticRegressionClassifier(IntentClassifier, GraphComponent):
+    """Intent classifier using the Logistic Regression."""
+
+    @classmethod
+    def required_components(cls) -> List[Type]:
+        """Components that should be included in the pipeline before this component."""
+        return [Featurizer]
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
+        return ["sklearn"]
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """The component's default config (see parent class for full docstring)."""
+        return {
+            "class_weight": "balanced",
+            "max_iter": 100,
+            "solver": "lbfgs",
+            "tol": 1e-4,
+            "random_state": 42,
+        }
+
+    def __init__(
+        self,
+        config: Dict[Text, Any],
+        name: Text,
+        model_storage: ModelStorage,
+        resource: Resource,
+    ) -> None:
+        """Construct a new classifier."""
+        self.name = name
+        self.clf = LogisticRegression(
+            solver=config["solver"],
+            max_iter=config["max_iter"],
+            class_weight=config["class_weight"],
+            tol=config["tol"],
+            random_state=config["random_state"],
+        )
+
+        # We need to use these later when saving the trained component.
+        self._model_storage = model_storage
+        self._resource = resource
+
+    def _create_X(self, messages: List[Message]) -> csr_matrix:
+        """This method creates a sparse X array that can be used for predicting."""
+        X = []
+        for e in messages:
+            # First element is sequence features, second is sentence features
+            sparse_feats = e.get_sparse_features(attribute=TEXT)[1]
+            # First element is sequence features, second is sentence features
+            dense_feats = e.get_dense_features(attribute=TEXT)[1]
+            together = hstack(
+                [
+                    csr_matrix(sparse_feats.features if sparse_feats else []),
+                    csr_matrix(dense_feats.features if dense_feats else []),
+                ]
+            )
+            X.append(together)
+        return vstack(X)
+
+    def _create_training_matrix(self, training_data: TrainingData):
+        """This method creates a scikit-learn compatible (X, y) training pairs."""
+        X = []
+        y = []
+
+        examples = [
+            e
+            for e in training_data.intent_examples
+            if (e.get("intent") and e.get("text"))
+        ]
+
+        for e in examples:
+            # First element is sequence features, second is sentence features
+            # for intent classification we're only interested in sentence features
+            sparse_feats = e.get_sparse_features(attribute=TEXT)[1]
```

Comment:
The extraction of `X` from an example looks like the same as in `_create_X`. Could this be merged into a helper function to be re-used in both places?